src/balancerd/src/lib.rs

```diff
@@ -673,6 +674,11 @@ impl mz_server_core::Server for PgwireBalancer {
         let outer_metrics = self.metrics.clone();
         let cancellation_resolver = self.cancellation_resolver.clone();
         let conn_uuid = Uuid::new_v4();
+        let peer_addr = conn
+            .peer_addr()
+            .ok()
+            .map(|a| a.ip().to_string())
+            .unwrap_or("unknown".into());
```

Comment:
The `unknown` here seems weird to me—seems likely that if we hit that case we'll forward on `unknown` to `environmentd`, which will then log an error about an unparseable IP in the proxy option. Probably better to just `?` return if the `peer_addr` is somehow missing.